---
title: "Problem Set 1"
author: "Gitanshu Munjal"
date: "Saturday, April 11, 2015"
output:
  pdf_document:
    highlight: tango
    keep_tex: yes
---

#Problem 1

##Calculate the allele frequenies for each locus.  
  
I define an R function to calculate allele frequencies for a bi-allelic locus given genotype frequencies. When provided genotype frequencies in the order (homozygous1, heterozygous, homozygous2), the function returns frequency of allele1, frequency of allele2, and 1-frequency of allele1 (added to double check calc. allele 2 frequency). I recycle this function for later questions in the assignment.
```{r}
################################################
# Allele Frequencies
################################################

allelefrequency <- function(ho1,het,ho2){
                                  a1 <- (2*ho1 + het)/(2*(ho1+het+ho2));
                                  a2 <- (2*ho2 + het)/(2*(ho1+het+ho2));
                                  a2c <- 1-a1;
                                  freqs <- c("allele1"=a1,"allele2"=a2,"1-allele1"=a2c)
                                  return(freqs)
                   }
  

#Locus1
ho1 <- cc <- 47     # allele1 = c
het <- ct <- 18
ho2 <- tt <- 35     # allele2 = t
allelefrequency(ho1,het,ho2)
```
**So,the frequency of the C and T alleles at Locus 1 are 0.56 and 0.44 respectively.**
```{r}
#Locus2
ho1 <- aa <- 50     # allele1 = a
het <- ag <- 42
ho2 <- gg <- 8      # allele2 = g
allelefrequency(ho1,het,ho2)
```
**So,the frequency of the A and G alleles at Locus 2 are 0.56 and 0.44 respectively.**  
\begin{center}
\line(1,0){250}
\end{center}
  
  
## What is the expected heterozygosity for each locus given the allele frequencies?
  
I define a function to return the expected frequency of heterozygous genotypes as 2pq when allele frequencies are p and q.
```{r}
################################################
# Expected heterozygosity
################################################

ehetero <- function(p,q){2*p*q}

#Locus1
ehetero(0.56,0.44)

#Locus2
ehetero(0.71,0.29)
```
**So, expected frequency of heterozygous genotypes at Locus 1 and Locus 2 are 0.4928 and 0.411 respectively.**
\begin{center}
\line(1,0){250}
\end{center} 
  
    
## Does the population significantly deviate from Hardy-Weinberg Equilibrium (HWE) at each locus?
```{r}
################################################
# Test for Hardy-Weinberg Equilibrium (HWE)
################################################

#Locus1
#Observed data (genotype counts)
occ <- 47     # allele1 = c
oct <- 18
ott <- 35     # allele2 = t
total <- occ+oct+ott

#Expected frequency (under HWE)
p <- as.numeric(allelefrequency(occ,oct,ott)[1])
q <- as.numeric(allelefrequency(occ,oct,ott)[2])

#Expected genotype counts
ecc <- p^2 * total
ect <- 2*p*q * total
ett <- q^2 * total
c("ecc"=ecc,"ect"=ect,"ett"=ett)

#Matrix containing observed and expected data
hwedat <- matrix(NA,3,3)
colnames(hwedat) <- c("observed","expected","(o-e)^2/e")
rownames(hwedat) <- c("cc","ct","tt")
hwedat[,1] <- c(occ,oct,ott)
hwedat[,2] <- c(ecc,ect,ett)
obs <- hwedat[,1]
exp <- hwedat[,2]
hwedat[,3] <- ((obs - exp)^2)/exp
hwedat

#Chi-square and p-value
chisq <- sum(hwedat[,3])
chisq                               #chi-squared value
pchisq(chisq,df=1,lower.tail=FALSE) #p-value
```
Indeed, **the population significantly deviates from HWE at Locus 1** at the 95\% confidence level.
  
  
```{r}

#Locus2
#Observed data (genotype counts)
oaa <- 50     # allele1 = a
oag <- 42
ogg <- 8     # allele2 = g
total <- oaa+oag+ogg

#Expected frequency (under HWE)
p <- as.numeric(allelefrequency(oaa,oag,ogg)[1])
q <- as.numeric(allelefrequency(oaa,oag,ogg)[2])

#Expected genotype counts
eaa <- p^2 * total
eag <- 2*p*q * total
egg <- q^2 * total
c("eaa"=eaa,"eag"=eag,"egg"=egg)

hwedat <- matrix(NA,3,3)
colnames(hwedat) <- c("observed","expected","(o-e)^2/e")
rownames(hwedat) <- c("aa","ag","gg")
hwedat[,1] <- c(oaa,oag,ogg)
hwedat[,2] <- c(eaa,eag,egg)
obs <- hwedat[,1]
exp <- hwedat[,2]
hwedat[,3] <- ((obs - exp)^2)/exp
hwedat

chisq <- sum(hwedat[,3])
chisq                               #chi-squared value
pchisq(chisq,df=1,lower.tail=FALSE) #p-value
```
At the 95\% confidence level, **the population does not significantly deviate from HWE at Locus 2.**
\begin{center}
\line(1,0){250}
\end{center}
  
    
## What are two possible reasons a population would deviate from HWE?
(1) Non-random mating 
(2) Migration (I/E-mmigration)
\begin{center}
\line(1,0){250}
\end{center}
  
\pagebreak

# Problem 2
## How many of each haplotype were definitively observed in this population?
  
All genotypes except the CTAG double heterozygote contribute towardss the observed haplotypes. Double homozygotes contribute $2N$ haplotypes and heterozygotes for one or the other locus contribute N haplotypes where $N$ is the number of individuals with the haplotype under consideration.
```{r}
################################################
# Definitively Observed Haplotypes
################################################

CA <- (2*40) + (2) + (4) 
TA <- (2*8) + (2) + (22)
CG <- (2*4) + (4) + (0)
TG <- (2*4) + (22) + (0)
Total <- CA + TA + CG + TG

#Matrix of Observed Haplotypes
obsdata <- matrix(NA,5,1)
colnames(obsdata) <- c("NumberObserved")
rownames(obsdata) <- c("CA","TA","CG","TG","Total")
obsdata[,1] <- c(CA,TA,CG,TG,Total)
obsdata
```
**The above matrix shows the number of each haplotype that were definitively observed.**
\begin{center}
\line(1,0){250}
\end{center}
  
    
##  Calculate D, D', and r2 for the C-A haplotype.
I use the definitive set of observed haplotypes to calculate allele frequencies. 
```{r}
################################################
# D, D', and r2 for the C-A haplotype
################################################

#Allele Frequencies
c <- (CA + CG) / Total
a <- (CA + TA) / Total
t <- (TA + TG) / Total
g <- (CG + TG) / Total
c("c"=c,"a"=a,"t"=t,"g"=g)
```
The allele frequencies are then used to calculate expected haplotype frequencies
```{r}
#Expected Haplotype Frequencies
expca <- c * a
expta <- t * a
expcg <- c * g
exptg <- t * g
c("expca"=expca,"expta"=expta,"expcg"=expcg,"exptg"=exptg)

#Data Matrix
parta <- matrix(NA,5,3)
colnames(parta) <- c("ObservedNumber","ObservedFrequency","ExpectedFrequency")
rownames(parta) <- c("CA","TA","CG","TG","Total")
parta[,1] <- c(CA,TA,CG,TG,Total)
parta[,2] <- parta[,1]/Total
parta[,3] <- c(expca,expta,expcg,exptg,expca+expta+expcg+exptg)
parta
```
```{r}
#D for C-A haplotype
obsca <- parta[1,2]
D <- obsca - expca
D
```
Thus, **D = 0.0744**
  
```{r}
#D prime for C-A haplotype
Dprime <- D / min(expcg,expta)
Dprime
```
Thus, **D' = 0.510**
  
```{r}
#R squared for C-A haplotype
Rsquared <- (D^2) / (c*a*t*g)
Rsquared
```
Thus, **R$^2$ = 0.121**
  
  
## Calculate Chi-square and determine if there is significant linkage disequilibrium (LD) at this locus.
  
We use expected and observed haplotype frequencies previously calculated to determine expected haplotype counts assuming the total allele count to be 168 and then compare these data with the observed data set to determine significance of LD using a chi-squared test.
```{r}
################################################
# Chi-square test to determine significance of LD
################################################

chisqtab <- matrix(NA,4,3)
colnames(chisqtab) <- c("observed","expected","(o-e)^2/e")
rownames(chisqtab) <- c("CA","TA","CG","TG")
obs <- chisqtab[,1] <- parta[1:4,2]*168
exp <- chisqtab[,2] <- parta[1:4,3]*168
chisqtab[,3] <- ((obs - exp)^2)/exp
chisqtab

chisq <- sum(chisqtab[,3])
chisq                               #chi-squared value
pchisq(chisq,df=1,lower.tail=FALSE) #p-value
```
Indeed, **there is significant LD at this locus** at the 95\% confience level.
\begin{center}
\line(1,0){250}
\end{center}
  
\pagebreak

# Problem 3
## Calculate the p-value of association between genotype and phenotype using a Chi-square test on expected and observed allele counts.
  
We recycle the function we defined in question 1 and use total genotype counts (across cases and controls) to determine allele frequencies.
```{r}
################################################
# Allele Frequencies
################################################

#Total counts
cc <- 150
ca <- 250
aa <- 100

#Allele Frequencies
allelefrequency(cc,ca,aa)[1] #allele1 = c ; allele2 = a)
c <- as.numeric(allelefrequency(cc,ca,aa)[1])
a <- as.numeric(allelefrequency(cc,ca,aa)[2])
```
Thus, the frequencies of the C and A allele are 0.55 and 0.45 respectively. We can use these frequencies to determine the expected allele counts for cases and controls by multiplying them each to the number of cases and controls respectively, as follows:
```{r}
#Matrix of observed allele counts
obsdata <- matrix(NA,2,2)
rownames(obsdata) <- c("C","A")
colnames(obsdata) <- c("Cases","Controls")
obsdata[,1] <- c(200,200)
obsdata[,2] <- c(350,250)
totalcasealleles <- sum(obsdata[,1])
totalcontrolalleles <- sum(obsdata[,2])
obsdata
```
\pagebreak
```{r}
#Matrix of expected allele counts
expdata <- matrix(NA,2,2)
rownames(expdata) <- c("C","A")
colnames(expdata) <- c("Cases","Controls")
expdata[,1] <- c(c,a)*totalcasealleles            #Expected case count
expdata[,2] <- c(c,a)*totalcontrolalleles         #Expected control count
expdata
```
Next, we use the above presented observed and expected matrices to determine significance of association using a chi-squared test.
```{r}
################################################
# Chi-square test for allele counts to determine association
################################################

x <- ((obsdata - expdata)^2) / expdata
chisq <- sum(x)
chisq                               #chi-squared value
pchisq(chisq,df=1,lower.tail=FALSE) #p-value
```
It would appear that the genotype is indeed significantly associated with the phenotype at the 95\% confidence level **(p-value: 0.009459189)**
\begin{center}
\line(1,0){250}
\end{center}
  
  
## If the researchers used a SNP chip with 100,000 SNPs. How many of these SNPs would you expect to be that significant by chance?
```{r}
pchisq(chisq,df=1,lower.tail=FALSE)*100000
```
I would expect 946 SNPs to be that significant just by chance.
\begin{center}
\line(1,0){250}
\end{center}
  
    
## Do you think this locus plays an important role in gluten sensitivity?
**No, given the p-value and the number of SNPs used I do not feel very confident** in the association of this locus with gluten sensitivity as even using the most naive adjustment (like Bonferroni) for the p-value, I would need to see a p-value much lower (to the order of p < 10^-6) to be fairly confident in the association.
  
# Problem 4  
```{r}
################################################
# Characterizing subpopulation structure
################################################

#Raw data provided
rawdata <- matrix(NA,3,6)
colnames(rawdata) <- c("SNP","RA","SP1","SP2","SP3","SP4")
rownames(rawdata) <- c("L1","L2","L3")
rawdata[1,] <- c("A/T","A",0.8,0.7,0.9,0.2)
rawdata[2,] <- c("G/C","G",0.6,0.1,0.8,0.7)
rawdata[3,] <- c("C/A","C",0.7,0.8,0.2,0.6)
rawdata <- as.table(rawdata)
rawdata

#Reduced raw data
obsdata <- matrix(NA,3,4) 
colnames(obsdata) <- c("SP1","SP2","SP3","SP4")
rownames(obsdata) <- c("L1","L2","L3")
obsdata[,1:4] <- as.numeric(rawdata[,3:6])
obsdata

#Expected heterozygosity matrix
exphs <- matrix(NA,3,4)
colnames(exphs) <- c("SP1","SP2","SP3","SP4")
rownames(exphs) <- c("L1","L2","L3")
exphs <- 2 * (obsdata) * (1 - obsdata)
exphs
```
\pagebreak  
      
## What is $F_{ST}$ between SP01 and SP03 at locus 1?
```{r}
hsbar <- rowMeans(subset(exphs, select = c(SP1, SP3)))[1]
averagep <- rowMeans(subset(obsdata, select = c(SP1, SP3)))[1]
ht <- 2 * averagep * (1-averagep)
loc1fst13 <- (ht -hsbar)/ht
loc1fst13
```
Thus, **$F_{ST}$ between SP01 and SP03 at locus 1 is 0.0196**
\begin{center}
\line(1,0){250}
\end{center}
  
    
## What is $F_{ST}$ between SP02 and SP04 at locus 2?
```{r}
hsbar <- rowMeans(subset(exphs, select = c(SP2, SP4)))[2]
averagep <- rowMeans(subset(obsdata, select = c(SP2, SP4)))[2]
ht <- 2 * averagep * (1-averagep)
loc2fst24 <- (ht -hsbar)/ht
loc2fst24
```
Thus, **$F_{ST}$ between SP02 and SP04 at locus 2 is 0.375**
\begin{center}
\line(1,0){250}
\end{center}
    
##  What is $F_{ST}$ among all four subpopulations at locus 3
```{r}
hsbar <- rowMeans(exphs)[3]
averagep <- rowMeans(obsdata)[3]
ht <- 2 * averagep * (1-averagep)
loc3fst <- (ht -hsbar)/ht
loc3fst
```
Thus, **$F_{ST}$ among all populations at locus 3 is 0.212**
\begin{center}
\line(1,0){250}
\end{center}
  
  
## What is the likelihood that this individual originated from SP01? What about SP02, SP03 and SP04?
The genotype AA GC CA translates to homozygosity for the reference allele at locus 1 and heterozygosity for the other two loci. If $p_{i}$ were the frequency of the ref. allele at $loc_{i}$ then the frequency of the observed genotype would be $(p_1^2)(2*p_2*(1-p_2))(2*p_3*(1-p_3))$

```{r}
probAA1 <- obsdata[1,]^2  
probGC2 <- 2 * obsdata[2,] * (1-obsdata[2,])
probCA3 <- 2 * obsdata[3,] * (1-obsdata[3,])
probAAGCCA <- probAA1 * probGC2 * probCA3
probAAGCCA
```
The above matrix shows the population-wise likelihoods for the genotype.
\begin{center}
\line(1,0){250}
\end{center}
\pagebreak  

## What is the posterior probability that the individual originated from SP01? What about SP02, SP03 and SP04?
```{r}
#Priors
priormatrix <- matrix(NA,4,1)
rownames(priormatrix) <- c("SP1","SP2","SP3","SP4")
priormatrix[,1] <- c(0.1,0.1,0.7,0.1)
colnames(priormatrix) <- c("priorprobability")
priormatrix

#Bayes Theorem Numerator
bayesnumerator <- probAAGCCA * priormatrix
colnames(bayesnumerator) <- c("bayesnumerator")
bayesnumerator

#Bayes Theorem Denominator
bayesdenominator <- sum(bayesnumerator)
bayesdenominator

#Posterior Probabilities
posteriorprob <- bayesnumerator/bayesdenominator
colnames(posteriorprob) <- c("posteriorprob")
posteriorprob
```
The above matrix shows the population wise posterior probability for the given genotype.
\begin{center}
\line(1,0){250}
\end{center}
  
  
# Problem 5
I wrote a python program to analyze the given sequence data. The program (also submitted) returns pairwise differences, s, $\pi$(rounded), and the sequences reduced to segregating sites only(plus 1st base). 
  
## How many segregating sites (s) are present in these data?
**10**
\begin{center}
\line(1,0){250}
\end{center}

## What is pi ($\pi$) in these data?
```{r}
Dij <- 66
n <- 6
pi <- Dij/((n*(n-1)) / 2)
pi
```
Thus, **$\pi$=4.4**
\begin{center}
\line(1,0){250}
\end{center}
  
## What are s and $\pi$ expressed in per site values?
```{r}
s <- 10
pi <- 4.4
bp <- 50

spersite <- s/(2*bp)
spersite
```
Thus, **s per site = 0.1.**
```{r}
pipersite <- pi/(2*bp)
pipersite
```
Thus, **$\pi$ per site = 0.044.**
\begin{center}
\line(1,0){250}
\end{center}
  
\pagebreak    

## What is the minor allele frequency spectrum for these data?
```{r}
sfsdata <- matrix(NA,4,2)
colnames(sfsdata) <- c("MinorAlleleFrequency","NumberofSites")
sfsdata[,2] <- c(40,5,4,1)
sfsdata[,1] <- c(0,1/6,2/6,3/6)
sfsdata

plot(sfsdata)
```
\pagebreak

## What is the derived allele frequency spectrum for these data?
```{r}
sfsdata <- matrix(NA,6,2)
colnames(sfsdata) <- c("DerivedAlleleFrequency","NumberofSites")
sfsdata[,2] <- c(40,4,2,1,2,1)
sfsdata[,1] <- c(0,1/6,2/6,3/6,4/6,5/6)
sfsdata

plot(sfsdata)
```



```{r}

```

```{r}

```

```{r}

```